code/game/objects/structures/popout_cake.dm

```diff
@@ -159,3 +175,18 @@
 	slices_amount -= Clamp(round(Proj.damage / 3), 1, 4)
 
 	check_slices()
+
+//When spawned, stuffs the corpse underneath it inside
+/obj/structure/popout_cake/corpse_grabber/New()
+	..()
+
+	initialize()
```

Comment:
Fuck actually honestly I have no real idea what'd fix it properly.

Just leave it like this would be my best guess, for these cakes it doesn't matter if it's called twice anyways.